homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,116 @@
+"""
+Support for Honeywell Lyric thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/lyric
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.lyric import LyricDeviceEntity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+from .const import DATA_LYRIC_CLIENT, DATA_LYRIC_DEVICES, DOMAIN
+
+DEPENDENCIES = ['lyric']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_SCAN_INTERVAL):
+        vol.All(vol.Coerce(int), vol.Range(min=1))
+})
+
+
+async def async_setup_entry(
+        hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Lyric sensor based on a config entry."""
+    lyric = hass.data[DOMAIN][DATA_LYRIC_CLIENT]
+
+    try:
+        devices = lyric.devices()
+    except Exception as exception:
+        raise PlatformNotReady from exception
+
+    hass.data[DOMAIN][DATA_LYRIC_DEVICES] = devices
+
+    devices = []
+    for location, device in lyric.devices():
+        if device.indoorTemperature:
+            devices.append(LyricSensor(
+                device, location, hass, 'indoorTemperature', 'Temperature',
+                hass.config.units.temperature_unit, 'mdi:thermometer'))
```

Comment:
Please set the entity `device_class` property appropriately when possible. See available device classes here:
https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes